habitat-lab/habitat/tasks/rearrange/social_nav/social_nav_sensors.py

```diff
@@ -496,25 +547,111 @@ def _get_sensor_type(self, *args, **kwargs):
         return SensorTypes.TENSOR
 
     def _get_observation_space(self, *args, config, **kwargs):
-        return spaces.Box(
-            shape=(1,),
-            low=np.finfo(np.float32).min,
-            high=np.finfo(np.float32).max,
-            dtype=np.float32,
-        )
+        if config.return_image:
+            return spaces.Box(
+                shape=(
+                    self._height,
+                    self._width,
+                    1,
+                ),
+                low=np.finfo(np.float32).min,
+                high=np.finfo(np.float32).max,
+                dtype=np.float32,
+            )
+        else:
+            return spaces.Box(
+                shape=(1,),
+                low=np.finfo(np.float32).min,
+                high=np.finfo(np.float32).max,
+                dtype=np.float32,
+            )
+
+    def _get_bbox(self, img):
+        """Simple function to get the bounding box, assuming that only one object of interest in the image"""
+        rows = np.any(img, axis=1)
+        cols = np.any(img, axis=0)
+        rmin, rmax = np.where(rows)[0][[0, -1]]
+        cmin, cmax = np.where(cols)[0][[0, -1]]
+        return rmin, rmax, cmin, cmax
 
     def get_observation(self, observations, episode, *args, **kwargs):
         found_human = False
         use_k = f"agent_{self.agent_id}_articulated_agent_arm_panoptic"
         if use_k in observations:
             panoptic = observations[use_k]
         else:
-            return np.zeros(1, dtype=np.float32)
+            if self._return_image:
+                return np.zeros(
+                    (self._height, self._width, 1), dtype=np.float32
+                )
+            else:
+                return np.zeros(1, dtype=np.float32)
+
+        if self._return_image:
+            tgt_mask = np.float32(panoptic == self._human_id)
+            if self._is_return_image_bbox:
+                # Get the bounding box
+                bbox = np.zeros(tgt_mask.shape)
+                if np.sum(tgt_mask) != 0:
+                    rmin, rmax, cmin, cmax = self._get_bbox(tgt_mask)
+                    bbox[rmin:rmax, cmin:cmax] = 1.0
+                return np.float32(bbox)
+            else:
+                return tgt_mask
+        else:
+            if (
+                np.sum(panoptic == self._human_id)
+                > self._human_pixel_threshold
+            ):
+                found_human = True
+
+            if found_human:
+                return np.ones(1, dtype=np.float32)
+            else:
+                return np.zeros(1, dtype=np.float32)
 
-        if np.sum(panoptic == self._human_id) > self._human_pixel_threshold:
-            found_human = True
 
-        if found_human:
-            return np.ones(1, dtype=np.float32)
+@registry.register_sensor
+class InitialGpsCompassSensor(UsesArticulatedAgentInterface, Sensor):
+    """
+    Get the relative distance to the initial starting location of the robot
```

Comment:
Clarify that for human agents this will return 0's.

Reply:
yes!